### PR TITLE
Add new entry on the Linux Kernel

### LINKED ISSUE #1271: [Entry] Open Source: Linux Kernel
### Reason/inspiration (optional)

A new entry on the Linux Kernel could be added to the [Open Source](https://www.codecademy.com/resources/docs/open-source) concept entry. Also it should be linked in the [Linus Torvalds](https://www.codecademy.com/resources/docs/general/linus-torvalds) and [Linux](https://www.codecademy.com/resources/docs/open-source/linux) pages. This entry could describe what the kernel is used for and how it fits in with the rest of the Linux architecture.

You can assign this issue to me if it's something you want to add. 😇 

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by YanisaHS**
### Description

I added the linux-kernel concept with linux-kernel.md file inside. The file:

1. Introduces the Linux kernel
2. Lists primary functions of the Linux kernel
3. Provides info on the usage of Linux kernel

I also linked this page to the [Linus Torvalds](https://www.codecademy.com/resources/docs/general/linus-torvalds) and [Linux](https://www.codecademy.com/resources/docs/open-source/linux) pages.

Closes #1271 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/open-source/concepts/linux-kernel/linux-kernel.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  - 'Linux'
+  - 'Open Source'
+  - 'Operating Systems'
+  - 'Unix'
+CatalogContent:
+  - 'paths/computer-science'
+  - 'paths/code-foundations'
+---
+
+The Linux kernel is the core program that runs the [Linux](https://www.codecademy.com/resources/docs/open-source/linux) operating system (OS). It’s the most important component of Linux and serves as a bridge between the software and hardware of Linux machines.
```
Comment (SSwiniarski) [inline content/open-source/concepts/linux-kernel/linux-kernel.md:18]:
```suggestion
The **Linux kernel** is the core program that runs the [Linux](https://www.codecademy.com/resources/docs/open-source/linux) operating system (OS). It’s the most important component of Linux and serves as a bridge between the software and hardware of Linux machines.
```

File: content/open-source/concepts/linux-kernel/linux-kernel.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Different processes have different priority levels, and higher priority processes typically get more resources. When a higher priority process requests more resources, these additional resources are given at the expense of lower priority processes. The Linux kernel is in charge of managing all this, and it allocates and deallocates resources to different processes as necessary.
+
+### Security and System Calls
+
+The Linux kernel is responsible for performing actions requested by system calls. A system call is a request from the user space to the OS or “kernel space”. All requests sent to the OS are system calls, including those exampled above.
+
+The system call design allows for separation of the user space and kernel space. This is because many risks exist in the user space, but more damage can be done to the system from the kernel space. This design provides additional security to the system, because it provides an additional layer between the user and the internal kernel.
+
+## Usage of the Linux Kernel
```
Comment (SSwiniarski) [inline content/open-source/concepts/linux-kernel/linux-kernel.md:46]: I'd take the text in this section and move it to the introductory paragraph on line 18, and delete this header.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@YanisaHS I did an initial review and only had a couple of comments.

**Comment by YanisaHS** [comment]
@SSwiniarski Thanks for your feedback!! Done! 🥳

File: content/open-source/concepts/linux-kernel/linux-kernel.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+---
+Title: 'Linux kernel'
```
Comment (yangc95) [inline content/open-source/concepts/linux-kernel/linux-kernel.md:2]:
```suggestion
Title: 'Linux Kernel'
```

File: content/open-source/concepts/linux-kernel/linux-kernel.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  - 'Operating Systems'
+  - 'Unix'
+CatalogContent:
+  - 'paths/computer-science'
+  - 'paths/code-foundations'
+---
+
+The **Linux kernel** is the core program that runs the [Linux](https://www.codecademy.com/resources/docs/open-source/linux) operating system (OS). It’s the most important component of Linux and serves as a bridge between the software and hardware of Linux machines.
+
+The Linux kernel is free and [open source](https://www.codecademy.com/resources/docs/open-source). It is derived from [Unix](https://www.codecademy.com/resources/docs/general/unix) and used in the Linux OS. Other Linux-derived kernels exist in operating systems, such as Android and Chrome OS.
```
Comment (yangc95) [inline content/open-source/concepts/linux-kernel/linux-kernel.md:20]:
```suggestion
The Linux kernel is free and [open source](https://www.codecademy.com/resources/docs/open-source) operating system kernel that is derived from [Unix](https://www.codecademy.com/resources/docs/general/unix). Other Linux-derived kernels exist in operating systems such as Android and Chrome OS.
```

File: content/open-source/concepts/linux-kernel/linux-kernel.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+## Functions of the Linux Kernel
+
+Some primary functions of the Linux kernel are listed below.
+
+### Hardware Interface
+
+The Linux kernel connects the Linux user space with the hardware. It interprets requests from applications and sends them to the appropriate hardware resources. This includes internal hardware, such as the [CPU](https://www.codecademy.com/resources/docs/general/cpu) or [RAM](https://www.codecademy.com/resources/docs/general/ram), and peripheral hardware, such as storage disks or monitors.
```
Comment (yangc95) [inline content/open-source/concepts/linux-kernel/linux-kernel.md:28]:
```suggestion
The Linux kernel connects the Linux user space with the hardware. It interprets requests from applications and sends them to the appropriate hardware resources. These include internal hardware, such as the [CPU](https://www.codecademy.com/resources/docs/general/cpu) or [RAM](https://www.codecademy.com/resources/docs/general/ram), and peripheral hardware, such as storage disks or monitors.
```

File: content/open-source/concepts/linux-kernel/linux-kernel.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+When an application needs to perform a task, it sends a request to the Linux OS. Then, the Linux kernel determines where to send the request. For example, if a user began listening to music on their computer, the Linux kernel would have to send this request to multiple hardware resources, including speakers or headphones.
```
Comment (yangc95) [inline content/open-source/concepts/linux-kernel/linux-kernel.md:30]:
```suggestion
When an application needs to perform a task, it sends a request to the Linux OS. Then, the Linux kernel determines where to send the request. For example, if a user began listening to music on their computer, the Linux kernel would have to send a request to multiple hardware resources, including speakers or headphones.
```

**Review by yangc95 — COMMENTED** [review]
@YanisaHS I had a few minor suggestions, but otherwise this is very well-written. Thanks and Happy New Year! Please tag me once you've finished looking over/resolving everything.

**Comment by YanisaHS** [comment]
Thank you for your feedback @yangc95 !! Done 🥳

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
@YanisaHS here is the live link for your entry: https://www.codecademy.com/resources/docs/open-source/linux-kernel

Thanks for you patience and consistent efforts 😄

cc: @SSwiniarski

**Pull request merged**